pokemongo_bot/cell_workers/sniper.py

```diff
@@ -0,0 +1,370 @@
+        last_position = self.bot.position[0:2]
+
+        # Teleport, so that we can see nearby stuff
+        self.bot.hb_locked = True
+        self._teleport_to(pokemon)
+
+        # If social is enabled and if no verification is needed, trust it. Otherwise, update encounter and spawnpoint IDs!
+        verify = not pokemon.get('encounter_id') or not pokemon.get('spawn_point_id')
+        exists = not verify and self.is_social
+
+        # If information verification have to be done, do so
+        if verify:
+            seconds_since_last_check = time.time() - self.last_cell_check_time
+
+            # Wait a maximum of MIN_SECONDS_ALLOWED_FOR_CELL_CHECK seconds before requesting nearby cells
+            if (seconds_since_last_check < self.MIN_SECONDS_ALLOWED_FOR_CELL_CHECK):
+                time.sleep(self.MIN_SECONDS_ALLOWED_FOR_CELL_CHECK - seconds_since_last_check)
+
+            nearby_pokemons = []
+            nearby_stuff = self.bot.get_meta_cell()
+            self.last_cell_check_time = time.time()
+
+            # Retrieve nearby pokemons for validation
+            nearby_pokemons.extend(nearby_stuff.get('wild_pokemons', []))
+            nearby_pokemons.extend(nearby_stuff.get('catchable_pokemons', []))
+
+            # Make sure the target really/still exists (nearby_pokemon key names are game-bound!)
+            for nearby_pokemon in nearby_pokemons:
+                nearby_pokemon_id = nearby_pokemon.get('pokemon_data', {}).get('pokemon_id', 0) or nearby_pokemon.get('pokemon_id', 0)
+
+                if nearby_pokemon_id == pokemon.get('id', -1):
+                    exists = True
+
+                    # Also, if the IDs arent valid, update them (nearby_pokemon key names are game-bound!)
+                    if not pokemon.get('encounter_id') or not pokemon.get('spawn_point_id'):
+                        pokemon['encounter_id'] = nearby_pokemon['encounter_id']
+                        pokemon['spawn_point_id'] = nearby_pokemon['spawn_point_id']
+                    break
+
+        # If target exists, catch it, otherwise ignore
+        if exists:
+            self._log('Meeting {} successful!'.format(pokemon['name']))
+            self._teleport_back_and_catch(last_position, pokemon)
+        else:
+            self._error('Could not meet {}. Reasons: too far, caught, expired or fake data. Skipping...'.format(pokemon['name']))
+            self._teleport_back(last_position)
+
+        # Save target and unlock heartbeat calls
+        self._cache(pokemon)
+        self.bot.hb_locked = False
+
+        return WorkerResult.SUCCESS
+
+    def work(self):
+        targets = []
+
+        # Retrieve the targets
+        if self.mode == SnipingMode.SOCIAL:
+            targets = self._get_pokemons_from_social()
+        elif self.mode == SnipingMode.URL:
+            targets = self._get_pokemons_from_url()
+
+        # Order the targets (descending)
+        for attr in self.order_by:
```

Comment:
It does sort by the attributes the user has specified, if thats what youre asking. So yes, it sorts correctly (tested).